db/crud.go

```diff
@@ -592,6 +592,13 @@ func (db *Database) ImportDoc(docid string, value []byte, isDelete bool) error {
 	}
 
 	_, err := db.updateDoc(docid, true, 0, func(doc *document) (Body, AttachmentData, error) {
+		// If this is a delete, and there is no xattr on the existing doc,
+		// we shouldn't import.  (SG purge arriving over DCP feed)
+		if isDelete && doc.CurrentRev == "" {
+			base.LogTo("Import+", "Import skipped for delete of doc with no existing SG xattr (SG purge): %s", docid)
+			return nil, nil, base.ErrImportCancelled
```

Comment:
I think it might be good to consolidate the terminology -- change the comment to "Import cancelled" or change the error name to `ErrImportSkipped`.  Another idea for a name would be `ErrImportIgnored`, since calling it "Skipped" leaves some uncertainty on whether it *should* have been skipped or not, whereas "Ignored" makes a stronger statement that the code knew what it was and decided it was not worth paying attention to.